lesson25/src/js/main.js

```diff
@@ -0,0 +1,54 @@
+import { toggleModal } from "./modal";
+import { validateForm } from "./validation";
+
+const agreeCheckBox = document.querySelector('[data-id="agree_check"]');
+const agreeButton   = document.querySelector('[data-id="agree_button"]');
+const submitButton  = document.querySelector('[data-id="submit_button"]');
+
+agreeButton.addEventListener('click' , () => {
+    agreeCheckBox.removeAttribute('disabled');
+    agreeCheckBox.classList.remove("field-invalid");
+    agreeCheckBox.checked = true;
+    changeDisabledStatusSubmitButton();
+});
+
+submitButton.addEventListener("click", (e) => {
+    e.preventDefault();
+    window.location.href = "register-done.html";
+});
+
+const options = {
+    root: document.querySelector('[data-id="modal-inner"]'),
+    threshold: 1
+};
+
+const removeDisabledForAgreeButton = ([entry]) => {
+    if(entry.isIntersecting){
+        const agreeButton  = document.querySelector('[data-id="agree_button"]');
+        agreeButton.removeAttribute('disabled');
+    }
+};
+
+const observer = new IntersectionObserver(removeDisabledForAgreeButton, options);
+observer.observe(document.querySelector('[data-id="last_text"]'));
+
+const invalidSelector = document.getElementsByClassName("field-invalid");
+const changeDisabledStatusSubmitButton = () => submitButton.disabled = !(invalidSelector.length === 0);
+
+const inputSelector = document.querySelectorAll('input');
+for (const input of inputSelector){
+    input.classList.add("field-invalid");
+
+    input.addEventListener("blur", () => {
+        validateForm(input);
+        changeDisabledStatusSubmitButton();
+    });
+
+    input.addEventListener("input" , () => {
+        input.classList.remove("field-invalid");
+        validateForm(input);
+        changeDisabledStatusSubmitButton();
+    });
```

Comment:
Q.
Can't it just be at blur?
Right now, every time I type a single character, all of this code is being executed,
and I felt that the performance was not good...(?)

Reply:
[Just Comment]
I felt that it was easier for the person looking at the input, because when the validation passes, it switches to whether the validate passed or not:eyes:

> every time I type a single character, all of this code is being executed,

But as Haru-san said, performance may not be good.:eyes:

Reply:
@haru-programming 
Thank you for your comment!

The reason why we are checking validation for every single character input is because we thought it would be less stressful for the user if we could check it in real time. 

In general forms, validation checks are often done at once by submitting the form, but in the specification of the challenge, the submit button cannot be pressed until everything is passed.

I'll have to go over the performance aspect again!

Reply:
@csakyo 
Thank you for your comment!
I'm going to have to do some digging to see if it's performing as bad as it could be!

Reply:
@haru-programming @csakyo 
Changed validation check on input event for password only. 

When performing real-time validation, it is better to check input fields where the correct answer is easy to find, such as name and email address, when they lose focus, and to check input fields where the correct answer is difficult to find, such as passwords, while they are being typed.
ref:
https://alistapart.com/article/inline-validation-in-web-forms/
https://www.bebit.co.jp/memo/2010/02/post_169.html

Changed object methods to functions. 
Sorry for the confusing code.
https://github.com/nakagawamai/js-lesson/pull/27/commits/b52ea1daa1edebffe4775eca32cd1fe6184edc6e

Reply:
ref very informative.
Thank you!
I thought the thing about checking only passwords with input was a good idea!
I have one additional comment on the operation check.
Please check:)